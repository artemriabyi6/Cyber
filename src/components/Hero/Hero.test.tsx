import React from 'react';
import { renderToString } from 'react-dom/server';
import HeroSection from './Hero';



describe('HeroSection Football Coach', () => {
  test('рендериться без помилок', () => {
    const htmlString = renderToString(<HeroSection />);
    expect(htmlString).toBeTruthy();
  });

  test('містить футбольну тематику', () => {
    const htmlString = renderToString(<HeroSection />);
    
    expect(htmlString).toContain('Стань кращим');
    expect(htmlString).toContain('футболістом');
    expect(htmlString).toContain('ІНДИВІДУАЛЬНІ ТРЕНУВАННЯ');
  });

  test('містить футбольні іконки та елементи', () => {
    const htmlString = renderToString(<HeroSection />);
    
    expect(htmlString).toContain('⚽');
    expect(htmlString).toContain('🥅');
    expect(htmlString).toContain('👟');
    expect(htmlString).toContain('🏆');
  });

  test('містить кнопки для тренувань', () => {
    const htmlString = renderToString(<HeroSection />);
    
    expect(htmlString).toContain('Розпочати тренування');
    expect(htmlString).toContain('Увійти в акаунт');
    expect(htmlString).toContain('href="/signup"');
    expect(htmlString).toContain('href="/signin"');
  });

  test('містить статистику тренера', () => {
    const htmlString = renderToString(<HeroSection />);
    
    expect(htmlString).toContain('50+');
    expect(htmlString).toContain('5 років');
    expect(htmlString).toContain('100%');
    expect(htmlString).toContain('учнів');
    expect(htmlString).toContain('досвіду');
    expect(htmlString).toContain('результат');
  });

  test('має футбольну кольорову схему', () => {
    const htmlString = renderToString(<HeroSection />);
    
    expect(htmlString).toContain('bg-linear-to-br from-green-900');
    expect(htmlString).toContain('text-yellow-400');
    expect(htmlString).toContain('bg-yellow-400');
  });

  test('відповідає snapshot', () => {
    const htmlString = renderToString(<HeroSection />);
    expect(htmlString).toMatchSnapshot();
  });
});