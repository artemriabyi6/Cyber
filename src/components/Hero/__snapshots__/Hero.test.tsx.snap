// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`HeroSection Football Coach –≤—ñ–¥–ø–æ–≤—ñ–¥–∞—î snapshot 1`] = `"<section class="min-h-screen bg-gradient-to-br from-green-900 via-green-800 to-emerald-900 flex items-center justify-center px-4 py-8 relative overflow-hidden"><div class="absolute inset-0 bg-black/20"></div><div class="absolute top-10 left-10 w-20 h-20 bg-white/10 rounded-full animate-pulse"></div><div class="absolute bottom-20 right-20 w-32 h-32 bg-white/5 rounded-full animate-pulse"></div><div class="absolute top-1/2 right-1/4 w-16 h-16 bg-yellow-400/20 rounded-full animate-bounce"></div><div class="max-w-7xl mx-auto w-full relative z-10"><div class="grid grid-cols-1 lg:grid-cols-2 gap-12 lg:gap-16 items-center"><div class="text-center lg:text-left"><div class="inline-flex items-center px-4 py-2 bg-yellow-400/20 rounded-full mb-6 border border-yellow-400/30"><span class="w-2 h-2 bg-yellow-400 rounded-full animate-ping mr-2"></span><span class="text-yellow-400 text-sm font-semibold">‚öΩ –Ü–ù–î–ò–í–Ü–î–£–ê–õ–¨–ù–Ü –¢–†–ï–ù–£–í–ê–ù–ù–Ø</span></div><h1 class="text-4xl lg:text-6xl font-bold leading-tight mb-6"><span class="bg-gradient-to-r from-white to-green-200 bg-clip-text text-transparent">–°—Ç–∞–Ω—å –∫—Ä–∞—â–∏–º</span><br/><span class="text-white">—Ñ—É—Ç–±–æ–ª—ñ—Å—Ç–æ–º</span><br/><span class="text-yellow-400">–≤–∂–µ —Å—å–æ–≥–æ–¥–Ω—ñ!</span></h1><p class="text-xl lg:text-2xl text-green-100 mb-8 leading-relaxed max-w-2xl">–Ü–Ω–¥–∏–≤—ñ–¥—É–∞–ª—å–Ω—ñ —Ç—Ä–µ–Ω—É–≤–∞–Ω–Ω—è –∑ –ø—Ä–æ—Ñ–µ—Å—ñ–π–Ω–∏–º —Ç—Ä–µ–Ω–µ—Ä–æ–º –¥–ª—è –ø—ñ–¥–≤–∏—â–µ–Ω–Ω—è —Ç–µ—Ö–Ω—ñ–∫–∏, —à–≤–∏–¥–∫–æ—Å—Ç—ñ —Ç–∞ –º–∞–π—Å—Ç–µ—Ä–Ω–æ—Å—Ç—ñ –≥—Ä–∏. –†–æ–∑–∫—Ä–∏–π —Å–≤—ñ–π –ø–æ—Ç–µ–Ω—Ü—ñ–∞–ª!</p><div class="flex flex-col sm:flex-row gap-4 mb-12 justify-center lg:justify-start"><a class="px-8 py-4 bg-gradient-to-r from-yellow-400 to-yellow-500 text-green-900 font-bold rounded-xl shadow-lg hover:shadow-xl transform hover:-translate-y-1 transition-all duration-300 hover:from-yellow-300 hover:to-yellow-400 flex items-center justify-center" href="/signup"><span class="mr-2">‚öΩ</span>–†–æ–∑–ø–æ—á–∞—Ç–∏ —Ç—Ä–µ–Ω—É–≤–∞–Ω–Ω—è</a><a class="px-8 py-4 bg-white/10 backdrop-blur-sm text-white font-semibold rounded-xl border-2 border-white/30 hover:bg-white/20 hover:border-white/50 transform hover:-translate-y-1 transition-all duration-300 flex items-center justify-center" href="/signin"><span class="mr-2">üë§</span>–£–≤—ñ–π—Ç–∏ –≤ –∞–∫–∞—É–Ω—Ç</a></div><div class="space-y-4 max-w-md mx-auto lg:mx-0"><div class="flex items-center gap-3 text-green-100"><div class="w-8 h-8 bg-yellow-400/20 rounded-full flex items-center justify-center flex-shrink-0 border border-yellow-400/30"><span class="font-bold text-yellow-400">‚öΩ</span></div><span class="text-lg">–Ü–Ω–¥–∏–≤—ñ–¥—É–∞–ª—å–Ω–∏–π –ø—ñ–¥—Ö—ñ–¥</span></div><div class="flex items-center gap-3 text-green-100"><div class="w-8 h-8 bg-yellow-400/20 rounded-full flex items-center justify-center flex-shrink-0 border border-yellow-400/30"><span class="font-bold text-yellow-400">üéØ</span></div><span class="text-lg">–¢–µ—Ö–Ω—ñ–∫–∞ —Ç–∞ —Ç–∞–∫—Ç–∏–∫–∞</span></div><div class="flex items-center gap-3 text-green-100"><div class="w-8 h-8 bg-yellow-400/20 rounded-full flex items-center justify-center flex-shrink-0 border border-yellow-400/30"><span class="font-bold text-yellow-400">üìä</span></div><span class="text-lg">–ê–Ω–∞–ª—ñ–∑ –ø—Ä–æ–≥—Ä–µ—Å—É</span></div></div><div class="flex flex-wrap gap-6 mt-8 justify-center lg:justify-start"><div class="text-center"><div class="text-2xl font-bold text-yellow-400">50+</div><div class="text-green-200 text-sm">—É—á–Ω—ñ–≤</div></div><div class="text-center"><div class="text-2xl font-bold text-yellow-400">5 —Ä–æ–∫—ñ–≤</div><div class="text-green-200 text-sm">–¥–æ—Å–≤—ñ–¥—É</div></div><div class="text-center"><div class="text-2xl font-bold text-yellow-400">100%</div><div class="text-green-200 text-sm">—Ä–µ–∑—É–ª—å—Ç–∞—Ç</div></div></div></div><div class="relative flex justify-center lg:justify-end"><div class="relative w-80 h-80 lg:w-96 lg:h-96 bg-white/10 backdrop-blur-sm rounded-2xl border border-white/20 shadow-2xl overflow-hidden"><div class="absolute inset-0 bg-green-700/30"><div class="absolute inset-0 bg-gradient-to-br from-green-600/20 to-emerald-800/20"></div><div class="absolute top-1/2 left-0 right-0 h-0.5 bg-white/30"></div><div class="absolute left-1/2 top-0 bottom-0 w-0.5 bg-white/30"></div><div class="absolute top-1/2 left-1/2 w-20 h-20 border-2 border-white/30 rounded-full -translate-x-1/2 -translate-y-1/2"></div></div><div class="absolute -top-6 -left-6 w-20 h-20 bg-yellow-400/20 backdrop-blur-sm rounded-2xl border border-yellow-400/30 flex items-center justify-center animate-float"><span class="text-2xl">ü•Ö</span></div><div class="absolute -bottom-8 -right-8 w-24 h-24 bg-white/15 backdrop-blur-sm rounded-2xl border border-white/30 flex items-center justify-center animate-float animation-delay-1000"><span class="text-3xl">üëü</span></div><div class="absolute top-1/2 -left-12 w-16 h-16 bg-yellow-400/20 backdrop-blur-sm rounded-2xl border border-yellow-400/30 flex items-center justify-center animate-float animation-delay-2000"><span class="text-2xl">‚öΩ</span></div><div class="absolute inset-0 flex items-center justify-center"><div class="text-center"><div class="w-24 h-24 lg:w-32 lg:h-32 bg-gradient-to-br from-yellow-400 to-orange-500 rounded-3xl flex items-center justify-center mx-auto mb-4 shadow-lg border-2 border-white/20"><span class="text-4xl lg:text-5xl">üèÜ</span></div><p class="text-white font-semibold text-lg bg-black/30 backdrop-blur-sm rounded-lg py-2 px-4">–¢–≤—ñ–π —à–ª—è—Ö –¥–æ —É—Å–ø—ñ—Ö—É</p></div></div><div class="absolute bottom-4 right-4 w-12 h-12 bg-white rounded-full shadow-lg animate-bounce flex items-center justify-center"><div class="w-8 h-8 bg-black rounded-full flex items-center justify-center"><div class="w-6 h-6 bg-white rounded-full"></div></div></div></div></div></div></div><div class="absolute bottom-0 left-0 right-0"><svg viewBox="0 0 1200 120" preserveAspectRatio="none" class="w-full h-12 text-green-950"><path d="M0,0V46.29c47.79,22.2,103.59,32.17,158,28,70.36-5.37,136.33-33.31,206.8-37.5C438.64,32.43,512.34,53.67,583,72.05c69.27,18,138.3,24.88,209.4,13.08,36.15-6,69.85-17.84,104.45-29.34C989.49,25,1113-14.29,1200,52.47V0Z" opacity=".25" fill="currentColor"></path><path d="M0,0V15.81C13,36.92,27.64,56.86,47.69,72.05,99.41,111.27,165,111,224.58,91.58c31.15-10.15,60.09-26.07,89.67-39.8,40.92-19,84.73-46,130.83-49.67,36.26-2.85,70.9,9.42,98.6,31.56,31.77,25.39,62.32,62,103.63,73,40.44,10.79,81.35-6.69,119.13-24.28s75.16-39,116.92-43.05c59.73-5.85,113.28,22.88,168.9,38.84,30.2,8.66,59,6.17,87.09-7.5,22.43-10.89,48-26.93,60.65-49.24V0Z" opacity=".5" fill="currentColor"></path><path d="M0,0V5.63C149.93,59,314.09,71.32,475.83,42.57c43-7.64,84.23-20.12,127.61-26.46,59-8.63,112.48,12.24,165.56,35.4C827.93,77.22,886,95.24,951.2,90c86.53-7,172.46-45.71,248.8-84.81V0Z" fill="currentColor"></path></svg></div></section>"`;
